src/App.tsx

```diff
@@ -1,7 +1,39 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string;
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    pressedKey: 'Nothing was pressed yet',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({
+        pressedKey: `The last pressed key is [${event.key}]`,
+      });
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({
+        pressedKey: `The last pressed key is [${event.key}]`,
+      });
+    });
+  }
```

Comment:
it doesnt work :)

function, that you pass to addEventListener and to removeEventListener are two different functions.

that is why your removeEventListener will not work

you should to put it to variable and pass variable to both methods
```suggestion
handlePressedKey = (event) => ...

  componentDidMount() {
    document.addEventListener('keyup', handlePressedKey);
  }

  componentWillUnmount() {
    document.removeEventListener('keyup', handlePressedKey);
  }
```

Reply:
@romanblyashuk94  Thanks for the explanation ! I fixed it